### PR TITLE
5 create tokenizer

### PR DESCRIPTION
**Pull request by flixm16**
Created all tokens for symbols and keywords used in the base language.

keyword token starts with "Key" and  ends with the Keyword itself. Example: "KeyInt" for "int"
Value token starts with "Val"
Operator token starts with "Op" 
Compiler function token starts with "Fun"

single symbols don't have a token and can be used with with single quotes

Every Keyword was successful recognized

### CODE REVIEW

**Comment by Servostar** [comment]
Operators either passed-through as character or token. Should be uniform -> pass Operator always as token

**Comment by Servostar** [comment]
Missing floating point literals, got integer and string

File: src/yacc/parser.y [hunk 1]
```diff
@@ -6,6 +6,60 @@
...
+
+%token KeyIn
+%token KeyOut
+%token KeyFun
+%token OpEquals
+%token OpAnd
+%token OpOr
+%token OpNot
+%token OpXor
+
```
Comment (Servostar) [inline src/yacc/parser.y:48]: Unnecessary line break

File: src/yacc/parser.y [hunk 2]
```diff
@@ -6,6 +6,60 @@
     extern int yylex();
 %}
 
+%union {
+    char *string;
```
Comment (Servostar) [inline src/yacc/parser.y:10]: Missing floating point

File: src/lex/lexer.l [hunk 3]
```diff
@@ -4,13 +4,96 @@
...
+"," return(',');
+";"  return(';');
+"==" return(OpEquals);
+"&&" return(OpAnd);
+"||" return(OpOr);
+"!!" return(OpNot);
+"^^" return(OpXor);
+"&" return('&');
+"^" return('^');
+"." return('.');
```
Comment (Servostar) [inline src/lex/lexer.l:78]: Operators should be uniform in types. Either pass as character OR as token

File: src/lex/lexer.l [hunk 4]
```diff
@@ -4,13 +4,96 @@
...
+"box" return(KeyBox);
+"#" return('#');
+"typeof" return(FunTypeof);
+"sizeof" return(FunSizeof);
+"filename" return(FunFilename);
+"funname" return(FunFunname);
+"lineno" return(FunLineno);
+"extsupport" return(FunExtsupport);
+
+
```
Comment (Servostar) [inline src/lex/lexer.l:92]: Sooo much whitespace

**Review by Servostar — CHANGES_REQUESTED** [review]
Missing floating point literal, make operator typing uniform, fix whitespace

File: src/lex/lexer.l [hunk 5]
```diff
@@ -6,11 +6,79 @@
...
+"filename" return(FunFilename);
+"funname" return(FunFunname);
+"lineno" return(FunLineno);
+"extsupport" return(FunExtsupport);
+
+[a-zA-Z_]+ { yylval.string = strdup(yytext); return(Ident); };
+[-+]?[0-9]+    { yylval.num = atoi(yytext); return(ValInt); };
+\"([^\"\n])*\" {yylval.string = strdup(yytext); return(ValStr);};
+\"\"\"([^\"\n]|\\\n)*\"\"\" {yylval.string = strdup(yytext); return(ValMultistr);};
+[-+]?[0-9]*\.[0-9]+ {yylval.real = atof(yytext); return(ValFloat);};
```
Comment (Servostar) [inline src/lex/lexer.l:82]: Remove sign in favor of a 'negate' operator. Meaning we shall implement no explicit sign into the language but instead negate a number in a expression instead.

File: src/yacc/parser.y [hunk 6]
```diff
@@ -6,6 +6,56 @@
...
+    int num;
+    float real;
```
Comment (Servostar) [inline src/yacc/parser.y:12]: Pass integer and floating point literals as string.
Let interpretation of bitwise represention and such loss to parser.

**Review by Servostar — APPROVED** [review]
Seems good

**Review by SirTalksalot75 — APPROVED** [review]

**Pull request merged**